datalad/support/gitrepo.py

```diff
@@ -2388,34 +2388,34 @@ def get_submodules_(self, paths=None):
             # below
             return
 
-        # taken from 3.9's pathlib, can be removed once the minimally
-        # supported python version
-        def is_relative_to(self, *other):
-            """Return True if the path is relative to another path or False.
-            """
-            try:
-                self.relative_to(*other)
-                return True
-            except ValueError:
-                return False
-
+        posix_mod_paths = [m.relative_to(self.pathobj).as_posix() for m in modinfo]
         if paths:
-            # ease comparison
-            paths = [self.pathobj / p for p in paths]
-            # constrain the report by the given paths
-
-            modinfo = {
-                # modpath is absolute
-                modpath: modprobs
-                for modpath, modprobs in modinfo.items()
-                # is_relative_to() also match equal paths
-                if (any(is_relative_to(p, modpath) for p in paths) or
-                    any(is_relative_to(modpath, p) for p in paths))
-            }
+            # harmonize them into relative to the repository
+            posix_paths = []
+            for path in paths:
+                path = ut.PurePath(path)
+                if path.is_absolute():
+                    try:
+                        path = path.relative_to(self.pathobj)
+                    except ValueError as exc:
+                        lgr.debug(
+                            "Path %s it not relative to %s, skipping since nothing should match it: %s",
```

Comment:
```suggestion
                            "Path %s it not underneath %s, skipping since nothing should match it: %s",
```

I think the phrasing is otherwise misleading. The issue is not that path is not relative to something (we know it's absolute).

Reply:
true, thanks!